crates/wasmparser/src/validator.rs

```diff
@@ -691,6 +823,68 @@ impl<'a> ValidatingParser<'a> {
+                        Err(e) => self.validation_error = Some(e),
+                    }
+                }
+            }
+            ParserState::BeginInstantiate { module, count } => {
+                if !self.config.operator_config.enable_module_linking {
+                    self.set_validation_error("module linking proposal not enabled");
+                } else if module as usize >= self.resources.module_type_indices.len() {
+                    self.set_validation_error("module is not defined");
+                } else if self.resources.instance_type_indices.len() >= MAX_WASM_INSTANCES {
+                    self.set_validation_error("instance count out of bounds");
+                } else {
+                    self.resources
+                        .instance_type_indices
+                        .push(InstanceDef::Instantiated { module_idx: module });
+                    let module_ty = self.resources.module_type_indices[module as usize];
+                    if count as usize != self.module_type_at(module_ty).unwrap().imports.len() {
+                        self.set_validation_error("wrong number of imports provided");
+                    } else {
+                        self.module_instantiation = Some((module_ty, 0));
+                    }
+                }
+            }
+            ParserState::InstantiateParameter { kind, index } => {
+                let (module_ty_idx, import_idx) = self.module_instantiation.take().unwrap();
+                let module_ty = self.module_type_at(module_ty_idx).unwrap();
+                let ty = module_ty.imports[import_idx].ty.clone();
+                match self.check_instantiate_field(&ty, kind, index) {
+                    Ok(()) => {
+                        self.module_instantiation = Some((module_ty_idx, import_idx + 1));
+                    }
+                    Err(e) => self.validation_error = Some(e),
+                }
+            }
+            ParserState::EndInstantiate => {
+                let (module_ty, import_idx) = self.module_instantiation.take().unwrap();
+                let module_ty = self.module_type_at(module_ty).unwrap();
+                if import_idx != module_ty.imports.len() {
+                    self.set_validation_error("not enough imports provided");
+                }
+            }
+            ParserState::AliasSectionEntry(ref alias) => match alias.instance {
+                AliasedInstance::Parent => {
+                    self.set_validation_error("parent instances not supported");
+                }
+                AliasedInstance::Child(instance_idx) => {
+                    let (kind, index) = (alias.kind, alias.index);
+                    match self.check_alias_entry(instance_idx, kind, index) {
+                        Ok(()) => {}
+                        Err(e) => self.validation_error = Some(e),
+                    }
+                }
+            },
             _ => (),
```

Comment:
`ParserState::InlineModule` is not listed, shall we TODO and/or error as not supported yet?

Reply:
Ah yes that's not explicitly handled yet, but it's something I hope to fix in the next PR.